src/docfx/moniker/MonikerRangeParser.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Docs.Build
+{
+    internal class MonikerRangeParser
+    {
+        private readonly EvaluatorWithMonikersVisitor _monikersEvaluator;
+
+        private MonikerRangeParser(EvaluatorWithMonikersVisitor monikersEvaluator)
+        {
+            _monikersEvaluator = monikersEvaluator;
+        }
+
+        public static (List<Error>, MonikerRangeParser) Create(IEnumerable<Moniker> monikers)
+        {
+            var (errors, evaluatorWithMonikersVisitor) = EvaluatorWithMonikersVisitor.Create(monikers);
+            return (errors, new MonikerRangeParser(evaluatorWithMonikersVisitor));
+        }
+
+        public (List<Error>, IEnumerable<string>) Parse(string rangeString)
+        {
+            var errors = new List<Error>();
+            IEnumerable<string> monikerNames = null;
+            try
+            {
+                var expression = ExpressionCreator.Create(rangeString);
+                monikerNames = expression.Accept(_monikersEvaluator);
+            }
+            catch (MonikerRangeException ex)
```

Comment:
replace `MonikerRangeException` with `Errors.InvalidMonikerRange().Throw()` ?

Reply:
`throw Errors.InvalidMonikerRange` to break execution, we should not proceed with a bad moniker range.